File: server/dto/pagination.go
package dto

type Pagination struct {
	Page            int    `json:"page"`
	Limit           int    `json:"limit"`
	Sort            string `json:"sort"`
	Total           int    `json:"total"`
	FirstPage       int    `json:"first_page"`
	LastPage        int    `json:"last_page"`
	PreviousPage    string `json:"previous_page"`
	NextPage        string `json:"next_page"`
	HasNextPage     bool   `json:"has_next_page"`
	HasPreviousPage bool   `json:"has_previous_page"`
	// Searchs      []Search    `json:"searchs"`
}

File: client/src/types/pagination.ts
export type OffsetPagination = {
  page: number;
  limit: number;
};

export type PaginationResponse<T> = {
  data: T;
  pagination: Pagination;
};

export type Pagination = {
  page: number;
  limit: number;
  total: number;
  totalPages: number;
  first: number;
  last: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
  nextPage: number;
  prevPage: number;
};
